plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -1049,6 +1050,9 @@ public Optional<ConnectorOutputMetadata> finishCreateTable(ConnectorSession sess
             // May happen when table has been already created concurrently.
             throw new TrinoException(TABLE_ALREADY_EXISTS, format("Table %s already exists", icebergTableHandle.name()), e);
         }
+        catch (RuntimeException e) {
```

Comment:
If the exception is during commit then you should change:
https://github.com/trinodb/trino/blob/232916b75d415a5eb643cf922492eb8513d99aae/plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java#L1620-L1624

Reply:
Did you mean here? https://github.com/trinodb/trino/blob/232916b75d415a5eb643cf922492eb8513d99aae/plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergUtil.java#L967

Reply:
No `commit` method only prepares commit
It's always used like:
https://github.com/trinodb/trino/blob/232916b75d415a5eb643cf922492eb8513d99aae/plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java#L1040-L1041
 most failures are from:
`commitTransaction(...) ->transaction.commitTransaction()`

Reply:
Not in this case, see the stack trace in the commit message. This is the relevant part:
```
        at org.apache.iceberg.SnapshotProducer.commit(SnapshotProducer.java:384)
        at io.trino.plugin.iceberg.IcebergUtil.commit(IcebergUtil.java:970)
        at io.trino.plugin.iceberg.IcebergMetadata.finishCreateTable(IcebergMetadata.java:1041)
```

Reply:
I refactored the error handling to keep it more consistent, @pajaks @wendigo @ebyhr PTAL